README.md
# CollectionShop

<a href="https://www.paypal.com/cgi-bin/webscr?hosted_button_id=37B3BPALKHXAS&amp;item_name=CollectionShop+(from+GitHub)&amp;cmd=_s-xclick">**Donate** using **PayPal** if you ❤ this addon!</a>

Find and buy the lowest buyout Mounts, Pets, Toys, Appearances, and Recipes.

*Incompatible with <a href="https://www.curseforge.com/wow/addons/auctioneer">Auctioneer</a> addon*

<img alt="Screenshot 1" src="Screenshot1.jpg" width="310" height="146">

**Patch 8.3.0 brought a major overhaul to the Auction House. CollectionShop's Live scan feature is temporarily disabled until it can be fixed.**

## Features
* Standalone addon that creates it's own tab on the Auction House for collection shopping.
* Displays models for Mounts, Pets, Appearances, and Recipes in Blizzard's Side Dress Up frame.
* Appearance categories (armor and weapons) and Recipes are for the current player.
* Perform a GetAll or Live scan of the Auction House.
* Buy All - Allows quickly buying all results from first to last (use with caution).

## Filters
* Quality - Common, Uncommon, Rare, Epic
* Category, Pet Family
* Not Collected for all modes
* Collected (1-2/3) and Collected (3/3) for Pets
* Collected - Unknown Sources for Appearances
* Collected - Known Sources for Appearances
* Pet Levels
* Required Player Level Ranges (Appearances)
* Group Pets By Species (uncheck to view all level, rarity, breed combos)
* Required Level above player's level
* Requires Profession beyond player's skill
* Requires Riding Skill (Mounts)
* Non-set Items (Appearances)
* Crafted by a Profession

## Options
* **Auctions Won Reminder** - Reminds you to use or equip auctions you've won after leaving CollectionShop
* **Max Item Price** - Discard auctions beyond a desired gold amount (mode specific)
* **Item Value Source** - <a href="https://www.curseforge.com/wow/addons/tradeskill-master">TradeSkillMaster</a> users can use a price source or custom price source to add the result column "% Item Value"
* **Auto-select After Auction Unavailable** - Toggles whether to select the next cheapest auction of the same appearance, same pet, etc. when an auction is no longer available.

## Bugs &amp; Suggestions
Please make sure you have the latest version and post into the Curse.com comments with details and if possible a screenshot. Thank you for taking the time to report an issue, it helps make the addon better for everyone.

## Known Issues
* After closing CollectionShop you must learn the auctions you won to exclude them from future Shop results.
* With the exception of recipe items in "Recipes" mode, items that require a specialization of a profession you've learned (e.g. Gnomish or Goblin Engineer for Engineers) won't be filtered out.
* Alt-Tabbing can reduce your scan speed due to Blizzard's "System &gt; Advanced &gt; Max Background FPS" setting.
* Does not consider connected realms, requires auction data for character's realm.
* **Pets** - A few require reputation with a faction or individual and won't be filtered out.
* Live scan is temporarily disabled until it can be fixed to work with patch 8.3.0+.

## Planned Features
* Fix Live scan feature.
* Filter for uncollected/collected breeds of Battle Pets (suggested by dsmania).

## Addons by cmc3215
* <a href="https://github.com/cmc3215/collectionshop">CollectionShop</a>
* <a href="https://github.com/cmc3215/professions-complete">Professions Complete</a>
* <a href="https://github.com/cmc3215/battle-net-friend-alert">Battle.net Friend Alert</a>
* <a href="https://github.com/cmc3215/hide-clean-up-bags-button">Hide Clean Up Bags Button</a>
* <a href="https://github.com/cmc3215/class-order-halls-complete">Class Order Halls Complete</a>
* <a href="https://github.com/cmc3215/war-campaigns-complete">War Campaigns Complete</a>
* <a href="https://github.com/cmc3215/work-orders-complete">Work Orders Complete</a>


Changelog.txt
**4.07**

- Release date: 12/11/2022
- Added LibAHTab for compatibility with TSM, Auctionator, and possible others which have begun using it in patch 10.x.x to prevent taint associated with using the former standard panel templates method of adding an AH tab.

**4.06**

- Release date: 12/04/2022
- Significant compatibility fixes for recent patches 10.0.0 and 10.0.2.
- Updated item and skills data.
- TOC updated for patch 10.0.2.

**4.05**

- Release date: 09/12/2022
- Minor compatibility fixes for recent patches.
- Updated item and skills data.
- TOC updated for patch 9.2.7.

**4.04**

- Release date: 10/19/2020
- Workaround added: Internal auction error following a game client AUCTION_HOUSE_BROWSE_FAILURE event that is a Blizzard bug was resetting CollectionShop after purchasing an auction.

**4.03**

- Release date: 10/14/2020
- Minor compatibility fixes for new expansions API changes.
- TOC updated for patch 9.0.1.

**4.02**

- Release date: 05/09/2020
- Bug fixed: Closing the DressUpFrame using the X at the top was not resetting the Shop Filters panel as it should.

**4.01**

- Release date: 05/09/2020
- Bug fixed: Battle Pet searches where a lower quality pet was cheapest but a higher quality was available was resulting in CollectionShop saying the auction was unavailable. This was because Blizzard's quality filters are somewhat broken for Battle Pets as the filter only works when selecting the highest quality the pet is available as. Workaround has been implemented.

**4.0**

- Release date: 05/08/2020
- TOC updated for patch 8.3.0.
- Live scan disabled temporarily until it can be made compatible with patch 8.3.0.
- Blizzard's full Auction House scan can be used as often as every 20 min now instead 15 min.
- Major compatibility fixes for patch 8.3.0.
- UI adjusted to match new Auction House appearance.
- New Crafted Gear added for latest patch.

**3.12**

- Release date: 12/21/2019
- Compatibility fixes for TSM 4 API changes. Removed support for TSM 3.

**3.11**

- Release date: 10/12/2019
- TOC updated for patch 8.2.5.

**3.10**

- Release date: 10/2/2019
- Additional compatibility fix for patch 8.2.5 that caused an LUA error.

**3.09**

- Release date: 10/1/2019
- Minor compatibility fixes for patch 8.2.5 that were causing LUA errors.
- Items updated for patch 8.2.5.

**3.08**

- Release date: 07/28/2019
- TOC updated for patch 8.2.
- New items added for patch 8.2.

**3.07**

- Release date: 12/20/2018
- TOC updated for patch 8.1.
- New items added for patch 8.1.

**3.06**

- Release date: 09/08/2018
- Increased timeout for querying the Auction House.

**3.05**

- Release date: 08/04/2018
- Fixed filter typo.

**3.04**

- Release date: 08/04/2018
- Bug fixed: Recipes mode was showing class-specific recipes the player was unable to learn.
- New "Requires Profession Specialization" filter added to Recipes mode. This will toggle whether to include recipes that the player possesses the profession for but not the specialization (e.g. Gnomish or Goblin Engineer).

**3.03**

- Release date: 08/02/2018
- Bug fixed: Lua errors and addon hanging were being caused by incorrect TSM 4 compatibility statements.

**3.02**

- Release date: 08/02/2018
- Bug fixed: Recipes mode Shop resulted in a Lua error. Required Level comparison was failing because table was being stored instead of number.

**3.01**

- Release date: 08/02/2018
- Bug fixed: "Requires Level" Shop Filter was no longer effective since changes were made in the previous release.

**3.0**

- Release date: 07/24/2018
- New "Recipes" mode for Profession Recipes
- New "Non-set items" filter for Appearances, unchecking it will show only items belonging to one or more of the Appearance sets.
- GetAll data for all realms will be wiped due to information required for the Recipes mode.
- Please be sure to close the World of Warcraft client completely before attempting to use the latest version of CollectionShop.

**2.11**

- Release date: 07/18/2018
- Minor code changes for TSM 4 Beta compatibility

**2.10**

- Release date: 07/18/2018
- TOC updated for patch 8.0.1.

**2.09**

- Release date: 07/05/2018
- Bug fixed: Toy box Shop resulted in Lua error.

**2.08**

- Release date: 07/05/2018
- Added item name filter.
- Added level filters to Appearances mode.
- Added "Crafted by a Profession" filter.
- Added tooltip to BuyAll button to briefly describe what it does.
- New items added for patch 8.0.1.

**2.07**

- Release date: 05/28/2018
- New option "Auto-select After Auction Unavailable." This option toggles automatically searching for and scrolling to the next cheapest auction of the same appearance, same pet, etc. when an auction is no longer available.
- New "Seller:" text box added that contains the seller name for the selected auction. The idea being that players can not only see who is selling the item, but also click into the box to copy the seller's name for contacting them via in-game mail.
- New warning message added concerning incompatibility with the Auctioneer addon.
- Bug fixed: The Live checkbox was being disabled after an item that was no longer available was removed without an additional item being selected.
- Added the game patch number that CollectionShop was created for to the Help tab.
- Updated Curse URL on Help tab.

**2.06**

- Release date: 10/01/2017
- Bug fixed: Four variables were being accidentally put into global namespace. For most users this had no effect, but for others it may have caused minor taint issues.

**2.05**

- Release date: 08/31/2017
- New items added for patch 7.3.0.
- Toys: Wormhole Generator: Argus
- Companion Pets: Model D1-BB-L3R, Naxxy

**2.04**

- Release date: 08/30/2017
- TOC Updated for patch 7.3
- Bug fixed: Lua error in patch 7.3 on PlaySound() API call.

**2.03**

- Release date: 08/04/2017

OptionsConfig.lua
--------------------------------------------------------------------------------------------------------------------------------------------
-- INIT
--------------------------------------------------------------------------------------------------------------------------------------------
local NS = select( 2, ... );
local L = NS.localization;
--------------------------------------------------------------------------------------------------------------------------------------------
-- CONFIG
--------------------------------------------------------------------------------------------------------------------------------------------
NS.options.cfg = {
	--
	mainFrame = {
		width		= 600,
		height		= 440,
		portrait	= true,
		frameStrata	= "MEDIUM",
		frameLevel	= "TOP",
		Init		= function( MainFrame )
			SetPortraitToTexture( MainFrame:GetName() .. 'Portrait', "Interface\\Icons\\MountJournalPortrait" );
		end,
		OnShow		= function( MainFrame )
			MainFrame:ClearAllPoints();
			MainFrame:SetPoint( "CENTER", 0, 0 );
		end,
		OnHide		= function( MainFrame ) end,
	},
	--
	subFrameTabs = {
		{
			-- Options
			mainFrameTitle	= NS.title,
			tabText			= OPTIONS,
			Init			= function( SubFrame )
				NS.TextFrame( "Description", SubFrame, string.format( L["These options apply to all characters on your account.\nMaking changes will interupt or reset %s Auction House scans."], NS.title ), {
					setPoint = {
						{ "TOPLEFT", "$parent", "TOPLEFT", 8, -8 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontRedSmall",
				} );
				NS.CheckButton( "AuctionsWonReminderCheckButton", SubFrame, L["Auctions Won Reminder"], {
					setPoint = { "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -18 },
					tooltip = string.format( L["Remind me to use or\nequip auctions I've won\nafter leaving %s."], NS.title ),
					db = "auctionsWonReminder",
				} );
				NS.TextFrame( "MaxItemPriceLabel", SubFrame, L["Max Item Prices"], {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 4, -8 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontHighlight"
				} );
				NS.TextFrame( "MaxItemPriceInfo", SubFrame, L["Hide auctions above this Item Price, 0 to show all."], {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -4 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontNormalSmall"
				} );
				for i = 1, #NS.modes do
					NS.InputBox( "MaxItemPriceMode" .. i .. "Editbox", SubFrame, {
						size = { 52, 20 },
						setPoint = { "TOPLEFT", ( i == 1 and "#sibling" or "$parentMaxItemPriceMode" .. ( i - 1 ) .. "Editbox" ), "BOTTOMLEFT", ( i == 1 and 4 or 0 ), -8 },
						numeric = true,
						maxLetters = 6,
						OnTabPressed = function()
							_G[SubFrame:GetName() .. "MaxItemPriceMode" .. ( ( i + 1 ) <= #NS.modes and ( i + 1 ) or 1 ) .. "Editbox"]:SetFocus();
						end,
						OnEnterPressed = function( self )
							self:ClearFocus();
						end,
						OnEditFocusGained = function( self )
							self:HighlightText();
						end,
						OnEditFocusLost = function( self )
							self:HighlightText( 0, 0 );
							if self:GetNumber() == 0 then
								self:SetNumber( 0 );
							end
						end,
						OnTextChanged = function( self, userInput )
							local copper = self:GetNumber() * 10000; -- Convert gold to copper
							if copper ~= NS.db["maxItemPriceCopper"][NS.modes[i]] then
								NS.db["maxItemPriceCopper"][NS.modes[i]] = copper;
								if NS.mode == NS.modes[i] and NS.IsTabShown() then
									NS.Reset(); -- Reset, option changed
								end
							end
						end,
					} );
					NS.TextFrame( "MaxItemPriceGoldIcon", SubFrame, "|TInterface\\MONEYFRAME\\UI-GoldIcon:14|t   " .. NS.modeColorCodes[i] .. NS.modeNames[i] .. FONT_COLOR_CODE_CLOSE, {
						size = { 200, 14 },
						setPoint = { "LEFT", "#sibling", "RIGHT", 3, 0 },
					} );
				end
				NS.TextFrame( "TSMItemValueSourceLabel", SubFrame, L["Item Value Source"], {
					setPoint = {
						{ "TOPLEFT", "$parentMaxItemPriceMode" .. #NS.modes .. "Editbox", "BOTTOMLEFT", -4, -12 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontHighlight",
				} );
				NS.TextFrame( "TSMItemValueSourceInfo", SubFrame, L["TradeSkillMaster price source or custom price source. For a list of price sources type /tsm sources."], {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -4 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontNormalSmall",
				} );
				NS.InputBox( "TSMItemValueSourceEditbox", SubFrame, {
					size = { 190, 20 },
					setPoint = { "TOPLEFT", "#sibling", "BOTTOMLEFT", 4, -4 },
					maxLetters = 50,
					OnEnterPressed = function( self )
						self:ClearFocus();
					end,
					OnEditFocusGained = function( self )
						self:HighlightText();
					end,
					OnEditFocusLost = function( self )
						self:HighlightText( 0, 0 );
						local source = strtrim( self:GetText() );
						if source ~= NS.db["tsmItemValueSource"] then
							-- Validate Source
							if TSM_API and source ~= "" then
								if not ( TSM_API and TSM_API.IsCustomPriceValid( source ) ) then
									NS.Print( RED_FONT_COLOR_CODE .. L["Not a valid price source or custom price source."] .. FONT_COLOR_CODE_CLOSE );
								end
							end
							--
							NS.db["tsmItemValueSource"] = source; -- Always update source regardless of validity
							NS.adjustScrollFrame = true; -- Always adjust when changed regardless of whether it's required
							if NS.IsTabShown() then
								NS.Reset(); -- Reset, option changed
							end
						end
					end,
				} );
				NS.TextFrame( "TSMItemValueSourceInfo2", SubFrame, L["(adds Item Value % column to results, leave blank to disable)"], {
					setPoint = {
						{ "LEFT", "#sibling", "RIGHT", 8, 0 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontNormalSmall",
				} );
				NS.CheckButton( "AutoselectAfterAuctionUnavailableCheckButton", SubFrame, L["Auto-select After Auction Unavailable"], {
					setPoint = { "TOPLEFT", "$parentTSMItemValueSourceEditbox", "BOTTOMLEFT", -8, -12 },
					tooltip = L["When not using BuyAll, this\noption will auto-select and scroll\nto the next cheapest auction of the\nsame appearance, same pet, etc.\n\nWhen using BuyAll, this option is\nignored because the next (first)\nauction is always auto-selected."],
					db = "autoselectAfterAuctionUnavailable",
				} );
			end,
			Refresh			= function( SubFrame )
				local sfn = SubFrame:GetName();
				_G[sfn .. "AuctionsWonReminderCheckButton"]:SetChecked( NS.db["auctionsWonReminder"] );
				for i = 1, #NS.modes do
					_G[sfn .. "MaxItemPriceMode" .. i .. "Editbox"]:SetNumber( NS.db["maxItemPriceCopper"][NS.modes[i]] / 10000 ); -- Convert copper to gold
				end
				_G[sfn .. "TSMItemValueSourceEditbox"]:SetText( NS.db["tsmItemValueSource"] );
				_G[sfn .. "AutoselectAfterAuctionUnavailableCheckButton"]:SetChecked( NS.db["autoselectAfterAuctionUnavailable"] );
			end,
		},
		{
			-- Buyouts
			mainFrameTitle	= NS.title,
			tabText			= L["Buyouts"],
			Init			= function( SubFrame )
				NS.Button( "NameColumnHeaderButton", SubFrame, NAME, {
					template = "CollectionShop_AuctionsWon_ColumnHeaderButtonTemplate",
					size = { 253, 19 },
					setPoint = { "TOPLEFT", "$parent", "TOPLEFT", -2, 0 },
				} );
				NS.Button( "ModeColumnHeaderButton", SubFrame, L["Mode"], {
					template = "CollectionShop_AuctionsWon_ColumnHeaderButtonTemplate",
					size = { 112, 19 },
					setPoint = { "TOPLEFT", "#sibling", "TOPRIGHT", -2, 0 },
				} );
				NS.Button( "ItemPriceColumnHeaderButton", SubFrame, L["Item Price"], {
					template = "CollectionShop_AuctionsWon_ColumnHeaderButtonTemplate",
					size = { 188, 19 },
					setPoint = { "TOPLEFT", "#sibling", "TOPRIGHT", -2, 0 },
				} );
				NS.Button( "RefreshButton", SubFrame, L["Refresh"], {
					size = { 96, 20 },
					setPoint = { "BOTTOMRIGHT", "#sibling", "TOPRIGHT", 2, 7 },
					fontObject = "GameFontNormalSmall",
					OnClick = function()
						SubFrame:Refresh();
						NS.Print( L["Buyouts Refreshed"] );
					end,
				} );
				NS.ScrollFrame( "ScrollFrame", SubFrame, {
					size = { 553, ( 26 * 11 - 5 ) },
					setPoint = { "TOPLEFT", "$parentNameColumnHeaderButton", "BOTTOMLEFT", 1, -3 },
					buttonTemplate = "CollectionShop_AuctionsWon_ScrollFrameButtonTemplate",
					update = {
						numToDisplay = 11,
						buttonHeight = 26,
						alwaysShowScrollBar = true,
						UpdateFunction = function( sf )
							local items = NS.auctionsWon;
							local numItems = #items;
							local sfn = SubFrame:GetName();
							FauxScrollFrame_Update( sf, numItems, sf.numToDisplay, sf.buttonHeight, nil, nil, nil, nil, nil, nil, sf.alwaysShowScrollBar );
							local offset = FauxScrollFrame_GetOffset( sf );
							for num = 1, sf.numToDisplay do
								local bn = sf.buttonName .. num; -- button name
								local b = _G[bn]; -- button
								local k = offset + num; -- key
								b:UnlockHighlight();
								if k <= numItems then
									_G[bn .. "_Name"]:SetText( ITEM_QUALITY_COLORS[items[k][4]].hex .. string.match( items[k][2], "%|h%[(.+)%]%|h" ) .. FONT_COLOR_CODE_CLOSE ); -- quality(4), itemLink(2)
									_G[bn .. "_Name"]:SetScript( "OnEnter", function( self )
										GameTooltip:SetOwner( self, "ANCHOR_TOP" );
										if string.match( items[k][2], "|Hbattlepet:" ) then -- itemLink(2)
											local _,speciesID,level,breedQuality,maxHealth,power,speed,customName = strsplit( ":", items[k][2] )
											BattlePetToolTip_Show( tonumber( speciesID ), tonumber( level ), tonumber( breedQuality ), tonumber( maxHealth ), tonumber( power ), tonumber( speed ), customName );
										else
											GameTooltip:SetHyperlink( items[k][2] );
										end
										b:LockHighlight();
									end );
									_G[bn .. "_Name"]:SetScript( "OnLeave", function() GameTooltip_Hide(); b:UnlockHighlight(); end );
									_G[bn .. "_ModeText"]:SetText( NS.modeColorCodes[items[k].modeNum] .. NS.modeNames[items[k].modeNum] .. FONT_COLOR_CODE_CLOSE );
									MoneyFrame_Update( bn .. "_ItemPriceSmallMoneyFrame", items[k][1] ); -- itemPrice(1)
									b:Show();
								else
									b:Hide();
								end
							end
						end
					},
				} );
				NS.TextFrame( "AuctionsWon", SubFrame, "", {
					size = { 553, 63 },
					setPoint = { "TOP", "#sibling", "BOTTOM" },
					fontObject = "GameFontNormal",
					justifyH = "CENTER",
				} );
			end,
			Refresh			= function( SubFrame )
				local sfn = SubFrame:GetName();
				--
				_G[sfn .. "ScrollFrame"]:Reset();
				_G[sfn .. "AuctionsWonText"]:SetText( string.format( L["%s\n%sBuyout tracking is reset when %s is closed|r"], #NS.auctionsWon .. " " .. GREEN_FONT_COLOR_CODE .. ( #NS.auctionsWon == 1 and L["Buyout"] or L["Buyouts"] ) .. FONT_COLOR_CODE_CLOSE .. ( NS.copperAuctionsWon > 0 and " (" .. NS.MoneyToString( NS.copperAuctionsWon, HIGHLIGHT_FONT_COLOR_CODE ) .. ")" or "" ), RED_FONT_COLOR_CODE, NS.title ) );
			end,
		},
		{
			-- GetAll Scan Data
			mainFrameTitle	= NS.title,
			tabText			= L["GetAll Scan Data"],
			Init			= function( SubFrame )
				NS.TextFrame( "NoDataFoundMessage", SubFrame, L["No GetAll scan data for any realms."], {
					setPoint = {
						{ "TOPLEFT", "$parent", "TOPLEFT", 8, -8 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontRedSmall",
				} );
				NS.TextFrame( "RealmLabel", SubFrame, L["Realm:"], {
					size = { 45, 16 },
					setPoint = { "TOPLEFT", "$parent", "TOPLEFT", 8, -8 },
				} );
				NS.DropDownMenu( "RealmDropDownMenu", SubFrame, {
					setPoint = { "LEFT", "#sibling", "RIGHT", -12, -1 },
					buttons = function()
						local t = {};
						for realmName,_ in pairs( NS.db["getAllScan"] ) do
							tinsert( t, { realmName .. " : " .. NS.SecondsToStrTime( time() - NS.db["getAllScan"][realmName]["time"] ) .. " " .. L["ago"], realmName } );
						end
						return t;
					end,
					width = 195,
				} );
				NS.Button( "DeleteDataButton", SubFrame, L["Delete Data"], {
					size = { 126, 22 },
					setPoint = { "BOTTOMRIGHT", "$parent", "BOTTOMRIGHT", -8, 8 },
					OnClick = function()
						StaticPopup_Show( "CS_REALM_DELETE", UIDropDownMenu_GetSelectedValue( _G[SubFrame:GetName() .. "RealmDropDownMenu"] ), NS.title, { ["realmName"] = UIDropDownMenu_GetSelectedValue( _G[SubFrame:GetName() .. "RealmDropDownMenu"] ) } );
					end,
				} );
				StaticPopupDialogs["CS_REALM_DELETE"] = {
					text = L["Delete GetAll scan data? %s\n\nThis will interupt or reset %s Auction House scans"];
					button1 = YES,
					button2 = NO,
					OnAccept = function ( self, data )
						-- Interupt Auction House scans
						if NS.scan.status == "scanning" then
							NS.Reset();
						end
						-- Delete
						NS.db["getAllScan"][data["realmName"]] = nil;
						collectgarbage( "collect" );
						NS.Print( RED_FONT_COLOR_CODE .. string.format( L["GetAll scan data deleted: %s"], data["realmName"] ) .. FONT_COLOR_CODE_CLOSE );
						-- Refresh
						SubFrame:Refresh();
						-- Reset
						if AuctionFrameCollectionShop and AuctionFrameCollectionShop:IsShown() then
							NS.Reset();
						end
					end,
					OnCancel = function ( self ) end,
					showAlert = 1,
					hideOnEscape = 1,
					timeout = 0,
					exclusive = 1,
					whileDead = 1,
				};
			end,
			Refresh			= function( SubFrame )
				local sfn = SubFrame:GetName();
				local firstRealm = next( NS.db["getAllScan"] );
				if firstRealm then
					_G[sfn .. "NoDataFoundMessage"]:Hide();
					_G[sfn .. "RealmLabel"]:Show();
					_G[sfn .. "RealmDropDownMenu"]:Show();
					_G[sfn .. "DeleteDataButton"]:Show();
					--
					_G[sfn .. "RealmDropDownMenu"]:Reset( firstRealm );
				else
					_G[sfn .. "NoDataFoundMessage"]:Show();
					_G[sfn .. "RealmLabel"]:Hide();
					_G[sfn .. "RealmDropDownMenu"]:Hide();
					_G[sfn .. "DeleteDataButton"]:Hide();
				end
			end,
		},
		{
			-- Help
			mainFrameTitle	= NS.title,
			tabText			= HELP_LABEL,
			Init			= function( SubFrame )
				NS.TextFrame( "Description", SubFrame, string.format( L["%s version %s for patch %s"], NS.title, NS.versionString, NS.releasePatch ), {
					setPoint = {
						{ "TOPLEFT", "$parent", "TOPLEFT", 8, -8 },
						{ "RIGHT", -8 },
					},
					fontObject = "GameFontRedSmall",
				} );
				NS.TextFrame( "SlashCommandsHeader", SubFrame, string.format( L["%sSlash Commands|r"], BATTLENET_FONT_COLOR_CODE ), {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -18 },
						{ "RIGHT", 0 },
					},
					fontObject = "GameFontNormalLarge",
				} );
				NS.TextFrame( "SlashCommands", SubFrame, string.format(
						L["%s/cs|r - Opens options frame to \"Options\"\n" ..
						"%s/cs buyouts|r - Opens options frame to \"Buyouts\"\n" ..
						"%s/cs getallscandata|r - Opens options frame to \"GetAll Scan Data\"\n" ..
						"%s/cs help|r - Opens options frame to \"Help\"\n" ..
						"%s/cs buyoutbuttonclick|r - Clicks the Buyout button on the Auction House tab.\n                                     Use in a Macro for fast key or mouse bound buying."],
						NORMAL_FONT_COLOR_CODE, NORMAL_FONT_COLOR_CODE, NORMAL_FONT_COLOR_CODE, NORMAL_FONT_COLOR_CODE, NORMAL_FONT_COLOR_CODE
					), {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -8 },
						{ "RIGHT", 0 },
					},
					fontObject = "GameFontHighlight",
				} );
				NS.TextFrame( "NeedMoreHelpHeader", SubFrame, string.format( L["%sNeed More Help?|r"], BATTLENET_FONT_COLOR_CODE ), {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -18 },
						{ "RIGHT", 0 },
					},
					fontObject = "GameFontNormalLarge",
				} );
				NS.TextFrame( "NeedMoreHelp", SubFrame, string.format(
						L["%sQuestions, Comments, Bugs, and Suggestions|r\n\n" ..
						"https://www.curseforge.com/wow/addons/collectionshop"],
						NORMAL_FONT_COLOR_CODE
					), {
					setPoint = {
						{ "TOPLEFT", "#sibling", "BOTTOMLEFT", 0, -8 },
						{ "RIGHT", 0 },
					},
					fontObject = "GameFontHighlight",
				} );
			end,
			Refresh			= function( SubFrame ) return end,
		},
	},
};
